### PR TITLE
Fix iphone x issues

### PR DESCRIPTION
**Pull request by SergioEstevao**
This PR solves layout issues on the iPhoneX for the legacy editor and the hybrid editor.

How to test:
 - Launch the demo app using the iPhoneX.
 - Scroll around on the demo content in visual mode and HTML mode.
 - Check if not overlap happens on the corners and bottom of the iPhone X
 - Check that making keyboard visible and invisible doesn't brake anything
 - Test the legacy editor for the same things.
 - Test on the older devices to see if nothing is broken.

### CODE REVIEW

**Comment by frosty** [comment]
Just running the demo app – the positioning of items looks better, but it seems like some of them are still getting squashed in portrait orientation:

screen shot 2017-10-09 at 14 53 42

This is on iPhone 8, iOS 11.0

**Comment by bummytime** [comment]
@SergioEstevao positioning does look good, but I am seeing a similar issue with the media button on the iPhone 8 (iPhone X looks good):

screen shot 2017-10-09 at 9 31 09 am

```
<UIImageView: 0x7fae44135750; frame = (0 0; 35 44); clipsToBounds = YES; opaque = NO; userInteractionEnabled = NO; 
```

**Comment by frosty** [comment]
fwiw, seeing the same on the legacy editor – here's one of the buttons:

`<UIImageView: 0x7ff817f3f030; frame = (0 0; 39 44); clipsToBounds = YES; opaque = NO; userInteractionEnabled = NO; layer = <CALayer: 0x60800043cc00>>`

**Comment by SergioEstevao** [comment]
@bummytime and @frosty ready for another look.

**Comment by frosty** [comment]
Latest changes definitely fix the 'flying toolbar', and the sizes of the items are more consistent. However, I noticed on an iPhone SE sized device, we lose the last toolbar item on iOS 11, which doesn't happen on iOS 10:

editor-bar-10-11

**Comment by SergioEstevao** [comment]
@frosty ready for another look.

File: Classes/WPEditorFormatbarView.m [hunk 1]
```diff
@@ -350,12 +359,12 @@ - (void)initBarButtonItem:(ZSSBarButtonItem*)barButtonItem
...
     customButton.selectedTintColor = self.selectedItemTintColor;
     customButton.disabledTintColor = self.disabledItemTintColor;
-    customButton.imageView.contentMode = UIViewContentModeCenter;
+    customButton.imageView.contentMode = UIViewContentModeScaleAspectFit;
     [customButton addTarget:target
                      action:selector
            forControlEvents:UIControlEventTouchUpInside];
     barButtonItem.customView = customButton;
-    barButtonItem.width = buttonImage.size.width;
+    barButtonItem.image = image;
```
Comment (frosty) [inline Classes/WPEditorFormatbarView.m]: How come we're setting the image on the bar button item? Isn't that not displayed if we're using a `customView`?

**Comment by frosty** [comment]
Same on iPhone 8 – notice the small image icon:

screen shot 2017-10-10 at 14 20 25

**Comment by SergioEstevao** [comment]
@frosty tweaked it a bit more. Have a look.

**Comment by bummytime** [comment]
@SergioEstevao The icon sizes look good and all display across multiple devices.

However, I noticed there was extra padding on the bottom of the format bar when the keyboard first displays (this happened on both iOS 10 & 11, multiple devices):

format_bar

The padding disappears when you dismiss the soft keyboard and then re-enable it.

**Comment by SergioEstevao** [comment]
@bummytime and @frosty ready for a final look

File: Classes/WPEditorFormatbarView.m [hunk 2]
```diff
@@ -92,6 +92,20 @@ - (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
...
+- (void)layoutSubviews {
+    [super layoutSubviews];
+    // HACK: Sergio Estevao (2017-10-10): Change the size of the items whe running on a device with a width smaller or equal than 320 (iPhone SE)
+    if (self.frame.size.width <= 320) {
+        for (UIBarButtonItem *item in self.leftToolbar.items) {
+            item.width = roundf(item.image.size.width * 0.75);
+        }
+    }
+
+    if( @available(iOS 11, *)) {
```
Comment (bummytime) [inline Classes/WPEditorFormatbarView.m]: @SergioEstevao For posterity, could you put a quick comment here about why this is used?

**Comment by bummytime** [comment]
@SergioEstevao really nice job keeping at this! 

On the iPhone X, when the soft keyboard is not visible, the format bar is not tap-able:

iphone_x_11

This does not appear to be the case on iPads with a hardware keyboard (even when the shortcuts & predictive text bars are turned off). Sooooo, I am not sure if this is a showstopper or not, but I thought I would point it out.

**Comment by bummytime** [comment]
Also, the bottom inset fix from your latest commit seems to have fixed that issue on both iOS 10 & 11! 😄

**Comment by SergioEstevao** [comment]
@bummytime thanks for keep testing it. So I did one more change that solves the problem you found out. Can you please give it a final look?

@frosty if you want to check it too?

**Comment by bummytime** [comment]
@SergioEstevao Your latest commit fixed all of the issues seen on the X (plus it still works everywhere else). 🎉 After addressing @frosty's comments above it's a :shipit: from me.

File: Classes/WPEditorView.m [hunk 3]
```diff
@@ -348,6 +350,22 @@ - (void)refreshInsetsForKeyboardOffset:(CGFloat)vOffset {
...
+    UIEdgeInsets insets = UIEdgeInsetsZero;
+    if (@available(iOS 11, *)) {
+        insets = self.sourceView.inputAccessoryView.safeAreaInsets;
+    }
+    CGFloat toolbarHeight = 42;
+    CGRect newFrame = CGRectMake(0, 0, self.bounds.size.width, toolbarHeight + insets.bottom);
+    self.sourceView.inputAccessoryView.frame = newFrame;
+    self.sourceViewTitleField.inputAccessoryView.frame = newFrame;
+    self.titleField.inputAccessoryView.frame = newFrame;
+    self.contentField.inputAccessoryView.frame = newFrame;
```
[the remark this answers is not in the document]
Comment (SergioEstevao) [reply Classes/WPEditorView.m]: They do, but I was just wary if they change in the future.

File: Classes/WPEditorView.m [hunk 4]
```diff
@@ -348,6 +350,22 @@ - (void)refreshInsetsForKeyboardOffset:(CGFloat)vOffset {
...
+    [self refreshInputViewsFrames];
+}
+
+- (void)refreshInputViewsFrames {
+
+    UIEdgeInsets insets = UIEdgeInsetsZero;
+    if (@available(iOS 11, *)) {
+        insets = self.sourceView.inputAccessoryView.safeAreaInsets;
+    }
+    CGFloat toolbarHeight = 42;
```
[the remark this answers is not in the document]
Comment (SergioEstevao) [reply Classes/WPEditorView.m]: Change it for a constant. intrinsicContentSize is returning zero...

**Comment by SergioEstevao** [comment]
@frosty changed the toolbar height to a constant. Ready for final approval!

**Pull request merged**